finality-aleph/src/sync/substrate/chain_status.rs

```diff
@@ -0,0 +1,147 @@
+use std::{
+    fmt::{Display, Error as FmtError, Formatter},
+    marker::PhantomData,
+};
+
+use aleph_primitives::{BlockNumber, ALEPH_ENGINE_ID};
+use log::warn;
+use sp_blockchain::Backend;
+use sp_runtime::{
+    generic::BlockId as SubstrateBlockId,
+    traits::{Block as BlockT, Header as SubstrateHeader},
+};
+
+use crate::{
+    justification::backwards_compatible_decode,
+    sync::{substrate::Justification, BlockStatus, ChainStatus, Header, LOG_TARGET},
+    AlephJustification,
+};
+
+/// What can go wrong when checking chain status
+#[derive(Debug)]
+pub enum Error<B: BlockT> {
+    MissingHash(B::Hash),
+    MissingJustification(B::Hash),
+}
+
+impl<B: BlockT> Display for Error<B> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
+        use Error::*;
+        match self {
+            MissingHash(hash) => {
+                write!(f, "no block for hash {:?}", hash)
```

Comment:
as far as i can see, this variant is returned only when there exists a block with this hash, but it is not available yet in our backend - maybe a subtle rephrasing will be helpful

Reply:
Yes, you are right. How about this message "Inconsistent state, no block for existing hash {:?}"? I could make something similar for justification.

Reply:
something like `data availability problem: no block for hash {hash:?}`? should be a bit more precise